### ISSUE TITLE
Nested records have unneeded import

### ISSUE THREAD

**Issue by frankbenoit** [eclipse-jdt/eclipse.jdt.ui#890]
Version: 2022-06 (4.24.0)
Build id: 20220609-1112

Given a record nesting a record:
````Java
record A( int v, B b ){
    record B( String n ){}
}
````
On organize imports, this will have an import for `B`. At the same time this `import` is marked as unneeded.

**Comment by jjohnstn** [comment]
Transferred as suggested changes are in UI and patch will be posted shortly.

**Comment by nlisker** [comment]
@jjohnstn I'm still seeing this (or a similar) bug:

```java
import com.test.Entry.Source.Bound;
import com.test.Entry.Source.Entity;
import com.test.Entry.Source.Innate;
import com.test.Entry.Source.Structural;

public record Entry() {

	public sealed interface Source permits Innate, Structural, Bound, Entity {
	
		record Innate() implements Source {} // error below would be on this line

		record Structural() implements Source {}

		record Bound() implements Source {}

		record Entity() implements Source {}
	}
}
```

with unused import warnings on all imports. Removing the import causes an error:

`The type Innate that implements a sealed interface Entry.Source should be a permitted subtype of Entry.Source`. This message is wrong I believe.

I'm using version: 2024-06 (4.32) | Build id: I20240601-0610.

**Comment by jjohnstn** [comment]
Hi @nlisker Your case above is a completely separate issue.  I tried the test with the up-coming 4.34 M1 (2024-12) and it works fine (no import warnings).  It may work with the soon to be released 4.33 (2024-09) which releases tomorrow.   If I remove the imports and then do an Organize Imports, it adds the imports with no warnings.  The original test works fine as well.
